runbotics/runbotics-scheduler/src/websocket/bot/bot-lifecycle.service.ts

```diff
@@ -1,62 +1,80 @@
 import { Injectable } from '@nestjs/common';
-import { IBot, ProcessInstanceEventStatus, ProcessInstanceStatus } from 'runbotics-common';
+import { IBot, IProcessInstance, IProcessInstanceEvent, ProcessInstanceEventStatus, ProcessInstanceStatus } from 'runbotics-common';
 import { ProcessInstanceService } from 'src/database/process-instance/process-instance.service';
 import { Logger } from 'src/utils/logger';
-import { BotProcessEventService } from './process-launch/bot-process-instance-event.service';
 import { ProcessInstanceEventService } from '#/database/process-instance-event/process-instance-event.service';
+import { BotProcessEventService } from './process-launch/bot-process-instance-event.service';
+import { BotProcessService } from './process-launch/bot-process-instance.service';
 
 @Injectable()
 export class BotLifecycleService {
     constructor(
         private readonly processInstanceService: ProcessInstanceService,
         private readonly processInstanceEventService: ProcessInstanceEventService,
         private readonly botProcessEventService: BotProcessEventService,
+        private readonly botProcessService: BotProcessService,
     ) { }
     
     private readonly logger = new Logger(BotLifecycleService.name);
 
-    async handleInterruptedProcessInstanceExecution(bot: IBot): Promise<void> {
-        const disconnectedBotProcessInstances = await this.processInstanceService.findAllByBotId(bot.id);
-        disconnectedBotProcessInstances.forEach(async (processInstance) => {
-            if (
-                processInstance.status !== ProcessInstanceStatus.INITIALIZING &&
-                processInstance.status !== ProcessInstanceStatus.IN_PROGRESS
-            ) return;
-
-            const newProcessInstance = { 
-                ...processInstance,
-                status: ProcessInstanceStatus.ERRORED,
-                error: 'Bot has been shut down',
-            };
+    async handleProcessInstanceInterruption(bot: IBot): Promise<void> {
+        const disconnectedInstance: IProcessInstance = await this.processInstanceService.findActiveByBotId(bot.id);
```

Comment:
change to find all

Reply:
done - still working despite many instances with status IN_PROGRESS (but does not fix the old ones)